airflow/triggers/base.py

```diff
@@ -0,0 +1,96 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Any, AsyncIterator, Dict, Tuple
+
+
+class BaseTrigger:
+    """
+    Base class for all triggers.
+
+    A trigger has two contexts it can exist in:
+
+     - As part of a DAG declaration, where it's declared.
```

Comment:
Is this true? What I've seen of the code so far Triggers only exist when they are created in side a task's execute function

Reply:
It's not, this is an older comment, will fix